### PR TITLE
Fix broken formatting

### PR DESCRIPTION
**Pull request by frederikprijck**
Previously, the formatting was broken.

This PR ensures the formatting is not broken anymore.

### CODE REVIEW

**Comment by lazarljubenovic** [comment]
I don't have enough hard disk space for these pointless characters. Please send me 100TB SSD external disk so I can merge this. Tnx!

File: src/nodes/ng-ast-node/component/component.ts [hunk 1]
```diff
@@ -1,14 +1,13 @@
-import { throwIfUndefined } from '../../../utils'
-import { Declarable } from '../declarable'
-import { ChangeDetectionStrategy, ViewEncapsulation } from './enums'
-import { Decorator, TypeGuards } from 'ts-morph'
-import * as tg from 'type-guards'
-import * as path from 'path'
-import { defaultTemplateConfig, Template } from '../template/template'
-import { LocationFile, LocationSpan } from '../location'
+import { throwIfUndefined } from '../../../utils';
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:1]: No.

File: src/nodes/ng-ast-node/component/component.ts [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
+import { Declarable } from '../declarable';
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:2]: Ugly.

File: src/nodes/ng-ast-node/component/component.ts [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
+import { ChangeDetectionStrategy, ViewEncapsulation } from './enums';
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:3]: Disgusting.

File: src/nodes/ng-ast-node/component/component.ts [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
+import { Decorator, TypeGuards } from 'ts-morph';
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:4]: Sickening.

File: src/nodes/ng-ast-node/component/component.ts [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
+import * as tg from 'type-guards';
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:5]: Outrageous.

File: src/nodes/ng-ast-node/component/component.ts [hunk 6]
```diff
@@ -1,14 +1,13 @@
...
+import * as path from 'path';
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:6]: Shameless.

File: src/nodes/ng-ast-node/component/component.ts [hunk 7]
```diff
@@ -1,14 +1,13 @@
...
+import { defaultTemplateConfig, Template } from '../template/template';
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:7]: Distasteful.

File: src/nodes/ng-ast-node/component/component.ts [hunk 8]
```diff
@@ -1,14 +1,13 @@
...
+import { LocationFile, LocationSpan } from '../location';
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:8]: Horrifying.

File: src/nodes/ng-ast-node/component/component.ts [hunk 9]
```diff
@@ -23,171 +22,234 @@ export class Component extends Declarable {
    *
    * @return The name of the selector, if it's defined. Otherwise, `undefined`.
    */
-  public getSelectorDefinition (): string | undefined {
+  public getSelectorDefinition(): string | undefined {
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:25]: This looks like a function call now. Do you really format your code like this?

File: src/nodes/ng-ast-node/component/component.ts [hunk 10]
```diff
@@ -23,171 +22,234 @@ export class Component extends Declarable {
...
     const property = this.getDecoratorProperty(
       'selector',
       TypeGuards.isStringLiteral,
-      `Expected @Component.selector to be a string literal`,
-    )
-    return property == null ? undefined : property.getLiteralValue()
+      `Expected @Component.selector to be a string literal`
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:29]: Please tell me you're religious so you can pray your sins away.

File: src/nodes/ng-ast-node/component/component.ts [hunk 11]
```diff
@@ -23,171 +22,234 @@ export class Component extends Declarable {
...
-    return throwIfUndefined(templateString, `Expected component to have an inline or external template.`)
+  public getTemplateLocationSpan(): LocationSpan {
+    const inlineTemplate = this.getInlineTemplateLocationSpan();
+    const externalTemplate = this.getExternalTemplateLocationSpan();
+    const templateString =
+      inlineTemplate != null ? inlineTemplate : externalTemplate;
+    return throwIfUndefined(
+      templateString,
+      `Expected component to have an inline or external template.`
+    );
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/component/component.ts:49]: What the fuck did you set column width to sqrt2 or something?

File: src/nodes/ng-ast-node/pipe/pipe.ts [hunk 12]
```diff
@@ -1,5 +1,3 @@
-import { Declarable } from '../declarable'
+import { Declarable } from '../declarable';
 
-export class Pipe extends Declarable {
-
-}
+export class Pipe extends Declarable {}
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/pipe/pipe.ts:3]: I just....

File: src/nodes/ng-ast-node/route.ts [hunk 13]
```diff
@@ -1,134 +1,141 @@
...
 export abstract class Route extends NgAstNode {
-
-  public constructor (project: Project,
-                      locationSpan: LocationSpan,
-                      private path: string) {
-    super(project, locationSpan)
+  public constructor(
+    project: Project,
+    locationSpan: LocationSpan,
+    private path: string
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/route.ts:13]: This is probably the only sane thing your automagical bullshit spitter will write.

File: src/nodes/ng-ast-node/template/factory.ts [hunk 14]
```diff
@@ -11,79 +11,178 @@ import {
...
+  const text = htmlNode.value;
+  return [
+    new TextTemplateNode(
+      project,
+      htmlNode.locationSpan,
+      htmlNode.tokens,
+      template,
+      text
+    )
+  ];
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/template/factory.ts:40]: Oh yes spread my codes like butter jam yesss

File: src/nodes/ng-ast-node/template/tokenizer/ast.ts [hunk 15]
```diff
@@ -6,213 +6,208 @@
...
-        path.push(ast)
+        path.push(ast);
       } else {
         // Returning a value here will result in the children being skipped.
-        return true
+        return true;
       }
     }
-  }
+  })();
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/template/tokenizer/ast.ts:208]: What the motherfucking fuck did you do here?

File: src/nodes/ng-ast-node/template/tokenizer/html_parser.ts [hunk 16]
```diff
@@ -6,16 +6,22 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {getHtmlTagDefinition} from './html_tags';
-import { Token, TokenizeOptions, TokenizeResult } from './lexer'
-import {ParseTreeResult, Parser} from './parser';
+import { getHtmlTagDefinition } from './html_tags';
+import { Token, TokenizeOptions, TokenizeResult } from './lexer';
+import { ParseTreeResult, Parser } from './parser';
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/template/tokenizer/html_parser.ts:11]: Oh God I had semis left here.

File: src/nodes/ng-ast-node/template/tokenizer/html_tags.ts [hunk 17]
```diff
@@ -6,107 +6,173 @@
...
-      ignoreFirstLf?: boolean
-    } = {}) {
+  constructor({
+    closedByChildren,
+    implicitNamespacePrefix,
+    contentType = TagContentType.PARSABLE_DATA,
+    closedByParent = false,
+    isVoid = false,
+    ignoreFirstLf = false
+  }: {
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/template/tokenizer/html_tags.ts:28]: At least you're better than whatever Google uses in their Angular repo. The most asymmetrical formatting I've ever seen is their clang config.

File: src/nodes/ng-ast-node/template/tokenizer/interpolation_config.ts [hunk 18]
```diff
@@ -7,41 +7,47 @@
...
-  /^[{}]$/,       // i18n expansion
-  /&(#|[a-z])/i,  // character reference,
-  /^\/\//,        // comment
-]
+  /^\s*$/, // empty
+  /[<>]/, // html tag
+  /^[{}]$/, // i18n expansion
+  /&(#|[a-z])/i, // character reference,
+  /^\/\// // comment
+];
```
Comment (lazarljubenovic) [inline src/nodes/ng-ast-node/template/tokenizer/interpolation_config.ts:15]: Would it be possible to mark this as "don't format"?

File: src/utils/array-destructing-resolver.spec.ts [hunk 19]
```diff
@@ -44,37 +49,46 @@ describe(`resolveArrayDestructing`, () => {
       const y2 = y1
       const y3 = y2
       const y4 = y3
-    `
+    `;
```
Comment (lazarljubenovic) [inline src/utils/array-destructing-resolver.spec.ts:52]: RRRRRRRRREEEEEEEEEEEE

File: src/utils/index.ts [hunk 20]
```diff
@@ -1,41 +1,54 @@
...
-export function throwIf<C> (guard: (x: any) => boolean): <T>(x: any, msg?: string) => Exclude<T, undefined>
-export function throwIf<C> (predicate: (x: any) => boolean) {
-  return <T> (x: T, msg?: string) => {
+export function throwIf<C>(
+  predicate: (x: any) => x is C
+): <T>(x: T, msg?: string) => Exclude<T, C>;
+export function throwIf<C>(
+  guard: (x: any) => boolean
+): <T>(x: any, msg?: string) => Exclude<T, undefined>;
+export function throwIf<C>(predicate: (x: any) => boolean) {
```
Comment (lazarljubenovic) [inline src/utils/index.ts:17]: You should seek professional medical help if you find this readable.

File: src/utils/metadata-json-reader.ts [hunk 21]
```diff
@@ -7,128 +7,136 @@ enum Symbolic {
...
-  MetadataSymbol_Constructor |
-  MetadataSymbol_Function |
-  MetadataSymbol_Select
+  | MetadataSymbol_Module
+  | MetadataSymbol_Class
+  | MetadataSymbol_Call
+  | MetadataSymbol_Reference
+  | MetadataSymbol_Constructor
+  | MetadataSymbol_Function
+  | MetadataSymbol_Select;
```
Comment (lazarljubenovic) [inline src/utils/metadata-json-reader.ts:89]: What the fuck is this not a syntax error?

**Review by lazarljubenovic — CHANGES_REQUESTED** [review]

**Comment by lppedd** [comment]
This PR would make a noticeable difference in my company workflow. Specifically we are developing a webapp for Scientology and Semicolon is a God to us.
Thanks @frederikprijck, you got a free Scientology pass, hit me on DM.

File: src/nodes/ng-ast-node/component/component.ts (hunk 1 above)
Comment (frederikprijck) [reply src/nodes/ng-ast-node/component/component.ts:1]: Resolved, please have a look at my updated changes.

File: src/nodes/ng-ast-node/component/component.ts (hunk 3 above)
Comment (frederikprijck) [reply src/nodes/ng-ast-node/component/component.ts:3]: Resolved, please have a look at my updated changes.

File: src/nodes/ng-ast-node/component/component.ts (hunk 2 above)
Comment (frederikprijck) [reply src/nodes/ng-ast-node/component/component.ts:2]: Resolved, please have a look at my updated changes.

File: src/nodes/ng-ast-node/component/component.ts (hunk 5 above)
Comment (frederikprijck) [reply src/nodes/ng-ast-node/component/component.ts:5]: Resolved, please have a look at my updated changes.

File: src/nodes/ng-ast-node/template/tokenizer/html_tags.ts (hunk 17 above)
Comment (frederikprijck) [reply src/nodes/ng-ast-node/template/tokenizer/html_tags.ts]: Resolved, please have a look at my updated changes.

File: src/nodes/ng-ast-node/template/tokenizer/interpolation_config.ts (hunk 18 above)
Comment (frederikprijck) [reply src/nodes/ng-ast-node/template/tokenizer/interpolation_config.ts:15]: Resolved, please have a look at my updated changes.

File: src/utils/array-destructing-resolver.spec.ts (hunk 19 above)
Comment (frederikprijck) [reply src/utils/array-destructing-resolver.spec.ts:52]: Resolved, please have a look at my updated changes.

File: src/utils/index.ts (hunk 20 above)
Comment (frederikprijck) [reply src/utils/index.ts:17]: Resolved, please have a look at my updated changes.

File: src/utils/metadata-json-reader.ts (hunk 21 above)
Comment (frederikprijck) [reply src/utils/metadata-json-reader.ts:89]: No.

File: src/nodes/ng-ast-node/component/component.ts (hunk 4 above)
Comment (frederikprijck) [reply src/nodes/ng-ast-node/component/component.ts:4]: Resolved, please have a look at my updated changes.

**Pull request closed without merging**